mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/impl/PojoMassIndexingDelegatingFailureHandler.java

```diff
@@ -39,4 +56,44 @@ public void handle(MassIndexingEntityFailureContext context) {
 		}
 		delegate.handle( builder.build() );
 	}
+
+	@Override
+	public void onFinish() {
+		// report other failures that went over the limits:
+		for ( Map.Entry<Object, NotRecordedFailure> entry : failureCounts.entrySet() ) {
+			NotRecordedFailure failure = entry.getValue();
+			long unreported = failure.count.get() - maxFailureCountPerOperation;
+			if ( unreported > 0 ) {
+				FailureContext.Builder builder = FailureContext.builder();
+				builder.throwable(
+						log.notReportedFailures(
+								entry.getKey(),
+								unreported,
+								failure.firstThrowable.getMessage(),
+								failure.firstThrowable
+						)
+				);
+				builder.failingOperation( entry.getKey() );
+				delegate.handle( builder.build() );
+			}
+		}
+	}
```

Comment:
I don't know how I feel about this.

On the one hand, you're right that people using a custom mass indexing failure handler probably shouldn't get that flooding protection by default, because they might rely on getting *all* errors.

On the other hand, they might want to opt-in, and with this solution, they can't. Conversely, they might want to opt out and keep using the default logging failure handler (the `delegate` in this class), and with this solution, they can't.

Another problem that bothers me: people setting a background failure handler in their configuration properties will get this flooding protection, but they might not want to.

To solve that, I think we need the following behavior:

* Move back the flooding protection logic to the notifier.
* Determine the actual limit based on a default provided by failure handlers, allowing users to override that through the `MassIndexer`:
  * The `FailureHandler` defines a `maxFailureCountPerOperation` through a getter.
  * The `MassIndexingFailureHandler` defines a `maxFailureCountPerOperation` through a getter.  `PojoMassIndexingDelegatingFailureHandler` just delegates to `FailureHandler`.
  * Those getters have a default implementation that returns `Long.MAX_VALUE`.
  * The getter of `LogFailureHandler` returns, say, 100.
  * By default we use the maximum provided by the failure handler, except when the user overrides it explicitly on the `MassIndexer`.

As to the signature of the getter... something like this?

```
/**
 * @param threshold The number of failures during one mass indexing beyond which the
 * failure handler will no longer be notified. This threshold is reached separately for each indexed type.
 * May be overridden by mass indexer parameters
 * (see {@code failureFloodingThreshold(long)} in the {@code MassIndexer} interface).
 */
default long failureFloodingThreshold() {
    return Long.MAX_VALUE;
}
```

And then in `MassIndexer`:

```
/**
 * @param threshold The number of failures during one mass indexing beyond which the
 * failure handler will no longer be notified. This threshold is reached separately for each indexed type.
 * Overrides the {@link MassIndexingFailureHandler#failureFloodingThreshold threshold defined by the failure handler itself}.
 * Defaults to {@code 100} with the default failure handler.
 * @return {@code this}, for method chaining.
 */
MassIndexer failureFloodingThreshold(long threshold);
```

WDYT?